shared/components/map.js

```diff
@@ -141,18 +160,29 @@ export default class Map extends Component {
       const newLocation = {
         latitude: position.coords.latitude,
         longitude: position.coords.longitude,
-        latitudeDelta: 0.0922,
-        longitudeDelta: 0.0421
+        latitudeDelta: 0.004724,
+        longitudeDelta: 0.004023
       };
       this.prevZoom = null;
       this.props.setLocation(newLocation);
+      const userLocation = {
+        latitude: position.coords.latitude,
+        longitude: position.coords.longitude
+      };
+      this.props.setUserLocation(userLocation);
+      this.setState({userLocationEnabled: true});
     },
     (error) => {
       // todo: handle this error when gps is off
       console.log(error);
+      this.setState({userLocationEnabled: false});
     });
   }
 
+  handleLocationButton() {

```

Comment:
good! And I think that function's name is not clear. How about getUserLocation or etc..